gui/src/videodecoder.cpp

```diff
@@ -55,6 +74,14 @@ VideoDecoder::VideoDecoder(HardwareDecodeEngine hw_decode_engine, ChiakiLog *log
 		avcodec_free_context(&codec_context);
 		throw VideoDecoderException("Failed to open codec context");
 	}
+	
+	
+	if(1)  //is RPi?
+	{	
+		pi_hw_decoder = new ChiakiPihwDecoder;
+		chiaki_pihw_decoder_init(pi_hw_decoder);
+	}
```

Comment:
Should this be some kind of define test?

Reply:
The idea is that the `ChiakiPihwDecoder` will be used **instead** of `VideoDecoder` rather than embedding it inside of it. Therefore, instead of your current `chiaki_pihw_decoder_get_buffer()`, you should implement a function that has exactly the signature `bool ChiakiVideoSampleCallback(uint8_t *buf, size_t buf_size, void *user);` where `void *user` will be used to pass the pointer to the `ChiakiPihwDecoder`.
Then, you can take this function and pass it here instead of `VideoSampleCb`: https://github.com/thestr4ng3r/chiaki/blob/8fbd1b942714e86c5173142080642d342c7bbfa3/gui/src/streamsession.cpp#L77
The decision whether to pass `VideoSampleCb` or the callback for the pi decoder should come from a combination of a define (whether the pi decoding is available at all) and a user setting (whether the user also wants to use it instead of ffmpeg).